depthai_ros_driver/src/dai_nodes/sensors/mono.cpp

```diff
@@ -0,0 +1,89 @@
+#include "depthai_ros_driver/dai_nodes/sensors/mono.hpp"
+
+#include "cv_bridge/cv_bridge.h"
+#include "depthai_bridge/ImageConverter.hpp"
+#include "image_transport/camera_publisher.hpp"
+#include "image_transport/image_transport.hpp"
+
+namespace depthai_ros_driver {
+namespace dai_nodes {
+Mono::Mono(const std::string& daiNodeName,
+           rclcpp::Node* node,
+           std::shared_ptr<dai::Pipeline> pipeline,
+           dai::CameraBoardSocket socket,
+           dai_nodes::sensor_helpers::ImageSensor sensor)
+    : BaseNode(daiNodeName, node, pipeline) {
+    RCLCPP_INFO(node->get_logger(), "Creating node %s", daiNodeName.c_str());
+    setNames();
+    monoCamNode = pipeline->create<dai::node::MonoCamera>();
+    ph = std::make_unique<param_handlers::MonoParamHandler>(daiNodeName);
+    ph->declareParams(node, monoCamNode, socket, sensor);
+    setXinXout(pipeline);
+    RCLCPP_INFO(node->get_logger(), "Node %s created", daiNodeName.c_str());
+};
+void Mono::setNames() {
+    monoQName = getName() + "_mono";
+    controlQName = getName() + "_control";
+}
+
+void Mono::setXinXout(std::shared_ptr<dai::Pipeline> pipeline) {
+    if(ph->getParam<bool>(getROSNode(), "i_publish_topic")) {
+        xoutMono = pipeline->create<dai::node::XLinkOut>();
+        xoutMono->setStreamName(monoQName);
+        monoCamNode->out.link(xoutMono->input);
+    }
+    xinControl = pipeline->create<dai::node::XLinkIn>();
+    xinControl->setStreamName(controlQName);
+    xinControl->out.link(monoCamNode->inputControl);
+}
+
+void Mono::setupQueues(std::shared_ptr<dai::Device> device) {
+    if(ph->getParam<bool>(getROSNode(), "i_publish_topic")) {
+        monoQ = device->getOutputQueue(monoQName, ph->getParam<int>(getROSNode(), "i_max_q_size"), false);
+        monoQ->addCallback(std::bind(&Mono::monoQCB, this, std::placeholders::_1, std::placeholders::_2));
+        monoPub = image_transport::create_camera_publisher(getROSNode(), "~/" + getName() + "/image_raw");
+        auto calibHandler = device->readCalibration();
+        monoInfo = dai::ros::calibrationToCameraInfo(calibHandler,
+                                                     static_cast<dai::CameraBoardSocket>(ph->getParam<int>(getROSNode(), "i_board_socket_id")),
+                                                     ph->getParam<int>(getROSNode(), "i_width"),
+                                                     ph->getParam<int>(getROSNode(), "i_height"));
+    }
+    controlQ = device->getInputQueue(controlQName);
+}
+void Mono::closeQueues() {
+    if(ph->getParam<bool>(getROSNode(), "i_publish_topic")) {
+        monoQ->close();
+    }
+    controlQ->close();
+}
+
+void Mono::monoQCB(const std::string& name, const std::shared_ptr<dai::ADatatype>& data) {
+    auto frame = std::dynamic_pointer_cast<dai::ImgFrame>(data);
+    cv::Mat cv_frame = frame->getCvFrame();
+    auto curr_time = getROSNode()->get_clock()->now();
```

Comment:
Can you setup a local time at the start of the device and use that as baseTime. and use getTimeStamp from the ImgFrame so that we are sending the timestamp from the sensor which is the middle of the exposure set.

Reply:
I added converters from depthai_bridge for publishing